cache/sampleStoreTimeout.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * ./cache/sampleStoreTimeout.js
+ *
+ * Timeout samples
+ */
+'use strict'; // eslint-disable-line strict
+const sampleStore = require('./sampleStore');
+const redisClient = require('./redisCache').client.sampleStore;
+const isTimedOut = require('../db/helpers/sampleUtils').isTimedOut;
+const constants = require('../api/v1/constants');
+const redisErrors = require('./redisErrors');
+const ONE = 1;
+
+/**
+ * Create sample array from responses using the samples count
+ * calculated before. The remaining are aspects. Create a hash of
+ * aspects, aspname -> asp object. When analyzing samples, get aspect
+ * from this hash using sample aspect name.
+ * @param  {Array} samples - Samples to be timedout
+ * @param  {Hash} aspects - Aspects hash, name-> object
+ * @param  {Date} curr - Current datetime
+ * @returns {Array} Commands needed to timeout samples
+ */
+function getSampleTimeoutCommands(samples, aspects, curr) {
+  const sampCmds = [];
+
+  for (let num = 0; num < samples.length; num++) {
+    const samp = samples[num];
+    const aspName = samp.name.split('|')[ONE];
+
+    if (!aspName) {
+      throw new redisErrors.ResourceNotFoundError({
+        explanation: 'Aspect not found.',
+      });
+    }
+
```

Comment:
When I just tried this in my console: `'abc'.split('|')[1]` and it returns undefined. Do you mean something else? I can still have a check though.